### PR TITLE
Added vetting and linting, and fixed all issues brought up.

### PR DESCRIPTION
**Pull request by mreddington**

### CODE REVIEW

File: lib/api.go [hunk 1]
```diff
@@ -34,16 +33,19 @@ import (
...
 )
 
-type Api struct {
+//API needs a comment
+type API struct {
 	Config Config
 }
 
-func (api Api) Noop(ctx *cli.Context) {
+//Noop needs a comment
```
Comment (dan9186) [inline lib/api.go]: This kind of defeats the purpose of turning on the linter to block the build if we intentionally short circuit it to shut it up.

File: lib/params.go [hunk 2]
```diff
@@ -120,36 +123,39 @@ func (params GetParams) String() string {
 	return urlValues.Encode()
 }
 
+//Generate needs a comment
 func (params GetParams) Generate(args []string, argConfigs []Arg) GetParams {
 	for index, arg := range args {
 		if argConfigs[index].Type != "object" && argConfigs[index].Type != "array" {
-			reflect.ValueOf(&params).Elem().FieldByName(strings.Replace(strings.Title(argConfigs[index].Name), "-", "", -1)).SetString(arg)
+			reflect.ValueOf(&params).Elem().FieldByName(getFieldByArgumentName(argConfigs[index].Name)).SetString(arg)
```
Comment (dan9186) [inline lib/params.go]: I'm curious, was this a suggested linting change?

File: lib/util.go [hunk 3]
```diff
@@ -36,37 +36,54 @@ import (
 )
 
 var (
-	Logger                           *logrus.Logger = logrus.New()
```
Comment (dan9186) [inline lib/util.go]: I didn't do an exhaustive check of these, but is it possible to get away with making these private?  That would get rid of the need to comment them.

File: lib/params.go (hunk 2 above)
Comment (mreddington) [reply lib/params.go]: This was a consequence of the linter. Lint wanted various fields renamed to conform to Go's convention and those changes were incompatible with how we were transforming these field parameters for reflection.

File: lib/util.go (hunk 3 above)
Comment (mreddington) [reply lib/util.go]: I'll look into that.

File: lib/api.go (hunk 1 above)
Comment (mreddington) [reply lib/api.go]: I've no idea how to write a good comment. What does the method do? It should be obvious by the name of the method or by looking at the code. Why do we need a Noop, etc? I'm cautious to say, as I'm very likely to get it wrong or be entirely irrelevant.

File: lib/util.go (hunk 3 above)
Comment (mreddington) [reply lib/util.go]: I was able to get rid of most all of them. I'd like to get rid of the debug variable entirely as that should be controlled by setting the logger level, and the debug printing to standard out should be printing debug statements - our logger is directed to standard out.

**Pull request merged**